### PR TITLE
Fix for #18843

### PR DESCRIPTION
**Pull request by inso**
| Q | A |
| --- | --- |
| Branch? | 2.3 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #18843 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

**Comment by inso** [comment]
ping @symfony/deciders

File: src/Symfony/Component/Yaml/Inline.php [hunk 1]
```diff
@@ -156,12 +156,22 @@ public static function dump($value, $exceptionOnInvalidType = false, $objectSupp
      */
     private static function dumpArray($value, $exceptionOnInvalidType, $objectSupport)
     {
+        if ($value) {
```
Comment (javiereguiluz) [inline src/Symfony/Component/Yaml/Inline.php]: I was intrigued by this problem: *"how to detect if an array is associative or not"* and I found two one-liners on StackOverflow. Maybe we can simplify this code?

```php
$isMapping = array_keys($value) !== range(0, count($value) - 1);

$isMapping = count(array_filter(array_keys($value), 'is_string')) > 0;
```

Comment (inso) [inline src/Symfony/Component/Yaml/Inline.php]: My solution uses less memory and is faster

**Comment by inso** [comment]
Benchmark:
 - result: http://inso.im/bench/is-mapping.txt
 - source: http://inso.im/bench/is-mapping.phps

**Comment by nicolas-grekas** [comment]
:+1:

File: src/Symfony/Component/Yaml/Inline.php (hunk 1 above)
Comment (xabbuh) [inline src/Symfony/Component/Yaml/Inline.php]: In the `Dumper` class we use `array_keys($input) !== range(0, count($input) - 1)` to detect the type of array. I am fine with changing the way we detect it, but then we should move the detection into a dedicated method an reuse it in both places.

**Comment by inso** [comment]
@xabbuh comments addressed, but I'm not sure where isHash method should be placed

File: src/Symfony/Component/Yaml/Inline.php [hunk 2]
```diff
@@ -146,6 +146,26 @@ public static function dump($value, $exceptionOnInvalidType = false, $objectSupp
     }
 
     /**
+     * Check if given array is hash or just normal indexed array.
+     *
+     * @param array $value The PHP array to check
+     *
+     * @return bool true if value is hash array, false otherwise
```
Comment (xabbuh) [inline src/Symfony/Component/Yaml/Inline.php]: The method should be marked as `@internal`.

File: src/Symfony/Component/Yaml/Tests/InlineTest.php [hunk 3]
```diff
@@ -169,6 +169,24 @@ public function testParseUnquotedAsteriskFollowedByAComment()
...
+     */
+    public function testIsHash($array, $expected)
+    {
+        $this->assertSame($expected, Inline::isHash($array));
+    }
+
+    public function getDataForIsHash()
+    {
+        return array(
+            array([], false),
```
Comment (xabbuh) [inline src/Symfony/Component/Yaml/Tests/InlineTest.php]: We cannot use the short array notation as we need to support PHP 5.3.

**Comment by inso** [comment]
@xabbuh done.

**Comment by nicolas-grekas** [comment]
Thank you @inso.

**Pull request closed without merging**